streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -117,7 +120,7 @@ public String queryableStoreName() {
     private KTable<K, V> doFilter(final Predicate<? super K, ? super V> predicate,
                                   final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal,
                                   final boolean filterNot) {
-        final String name = builder.newProcessorName(FILTER_NAME);
+        final String name = new NamedInternal(predicate).orElseGenerateWithPrefix(builder, FILTER_NAME);
```

Comment:
validation missed?

Reply:
Actually I'd suggest we make validation private and called internally (see my other comment on NamedInternal).